q2_types/per_sample_sequences/_format.py

```diff
@@ -137,21 +142,56 @@ def sequences_path_maker(self, sample_id, barcode_id, lane_number,
         return '%s_%s_L%03d_R%d_001.fastq.gz' % (sample_id, barcode_id,
                                                  lane_number, read_number)
 
+    def _find_duplicates(self, ids):
+        return {x for x, c in collections.Counter(ids).items() if c > 1}
+
+    def validate(self):
+        forwards = []
+        reverse = []
+        for p in self.path.iterdir():
+            if p.is_dir():
+                d = p.relative_to(self.path)
+                raise ValidationError("Contains a subdirectory: %s" % d)
```

Comment:
Since files that don't match the naming convention are ignored, it might be useful to ignore subdirectories too (for user convenience). I know this is different than what we originally discussed, just thought of it now while reviewing.

Alternatively, we could have the validator error if there are any extra files or subdirectories. Would extra files/subdirs get copied into the `.qza` during import?

Reply:
I think we only ignore dotfiles (maybe?) Otherwise, they aren't ignored, the `_validate_` will see it has extra paths that none of the File/FileCollections own and will error (that's always been the case)

Reply:
Oh, what's confusing about this is the fact that a path like: `Human_Kneecap/S1_L001_R1_001.fastq.gz` *does* match the pathspec, so the `_validate_` doesn't see any unaccounted for files (which is how this error gets through). We realized that it's probably easier to just check if there are any directories that got caught up in that, since we need to loop over the files anyways. This is definitely worth a comment.